internal/compiler/lint/rules/js/preferOptionalChaining.ts

```diff
@@ -0,0 +1,132 @@
+import {
+	AnyJSExpression,
+	JSMemberExpression,
+	JSReferenceIdentifier,
+	jsMemberExpression,
+} from "@internal/ast";
+import {createVisitor, signals} from "@internal/compiler";
+import {descriptions} from "@internal/diagnostics";
+
+function checkLeftOperand(expr: AnyJSExpression): JSReferenceIdentifier | null {
+	if (expr.type === "JSReferenceIdentifier") {
+		return expr;
+	}
+	if (
+		expr.type === "JSBinaryExpression" &&
+		["!=", "!=="].includes(expr.operator)
```

Comment:
There's only a couple of cases here so it's faster to just inline the checks, there's not a lot of repetition to justify it.

```suggestion
		(expr.operator === "!=" || expr.operator === "!==)
```